examples/common_vmseries_and_autoscale/variables.tf

```diff
@@ -0,0 +1,362 @@
+  - `accelerated_networking` : (bool|`null`- module defaults) enable Azure accelerated networking for all dataplane network interfaces
+  - `use_custom_image` : (bool|`false`) 
+  - `custom_image_id` : (string|reqquired when `use_custom_image` is `true`) absolute ID of your own Custom Image to be used for creating new VM-Series
+  - `application_insights_id` : (string|`null`) ID of Application Insights instance that should be used to provide metrics for autoscaling
+  - `interfaces` : (list(string)|`[]`) configuration of all NICs assigned to a VM. A list of maps, each map is a NIC definition. Notice that the order DOES matter. NICs are attached to VMs in Azure in the order they are defined in this list, therefore the management interface has to be defined first. Following properties are available:
+    - `name` : (string|required) string that will form the NIC name
+    - `subnet_key` : (string|required) a key of a subnet as defined in `var.vnets`
+    - `create_pip` : (bool|`false`) flag to create Public IP for an interface, defaults to `false`
+    - `load_balancer_key` : (string|`null`) key of a Load Balancer defined in the `var.loadbalancers` variable
+    - `application_gateway_key` : (string|`null`) key of an Application Gateway defined in the `var.appgws`
+    - `pip_domain_name_label` : (string|`null`) prefix which should be used for the Domain Name Label for each VM instance
+  - `autoscale_config` : (map|`{}`) map containing basic autoscale configuration
+    - `count_default` : (number|`null` - module defaults) default number or instances when autoscalling is not available
+    - `count_minimum` : (number|`null` - module defaults) minimum number of instances to reach when scaling in
+    - `count_maximum` : (number|`null` - module defaults) maximum number of instances when scaling out
+    - `notification_emails` : (list(string)|`null` - module defaults) a list of e-mail addresses to notify about scaling events
+  - `autoscale_metrics` : (map|`{}`) metrics and thresholds used to trigger scaling events, see module documentation for details
+  - `scaleout_config` : (map|`{}`) scale out configuration, for details see module documentation
+    - `statistic` : (string|`null` - module defaults) aggregation method for statistics coming from different VMs
+    - `time_aggregation` : (string|`null` - module defaults) aggregation method applied to statistics in time window
+    - `window_minutes` : (string|`null` - module defaults) time windows used to analyze statistics
+    - `cooldown_minutes` : (string|`null` - module defaults) time to wait after a scaling event before analyzing the statistics again
+  - `scalein_config` : (map|`{}`) scale in configuration, same properties supported as for `scaleout_config`
+
+  Example, no auto scaling:
+
+  ```
+  {
+  "vmss" = {
+    name              = "ngfw-vmss"
+    vnet_key          = "transit"
+    bootstrap_options = "type=dhcp-client"
```

Comment:
there's a bug in the dedicated vmss: `"type=dhcp"`, maybe we could use this PR to fix it there

Reply:
Yes saw that, I was going to open a separate PR for that but let me update this then.